feat-paysys: Define New API Endpoints

fatimali03 opened the pull request:
# SPDX-License-Identifier: Apache-2.0

## What did we change?

## Why are we doing this?

## How was it tested?
- [x] Locally
- [x] Development Environment
- [x] Husky successfully run
- [x] Unit tests passing

fatimali03 committed:
feat: basic setup

MuhammadAli-Paysys committed:
feat: Knex Configuration

UmairKhan-Paysys committed:
feat: boilerplate

fatimali03 committed:
Merge pull request #2 from tazama-lf/feat-paysys-knex

Feat paysys knex

fatimali03 committed:
feat: implement backend for API endpoint definition, schema inference, validation, workflow, and audit logging

fatimali03 committed:
Merge pull request #3 from tazama-lf/feat-paysys-1.1

feat: implement backend for API endpoint definition, schema inference…

fatimali03 committed:
Revert "feat: implement backend for API endpoint definition, schema inference…"

fatimali03 committed:
Merge pull request #4 from tazama-lf/revert-3-feat-paysys-1.1

Revert "feat: implement backend for API endpoint definition, schema inference…"

fatimali03 committed:
feat: Field Mapping and Data Transformation

fatimali03 committed:
fix: comment for checking

fatimali03 committed:
Merge: Preserve all feat-paysys-1.1 functionality and files

fatimali03 committed:
fix: lint fixes

Abdullah-Siddiqui-Paysys committed:
feat: pushed frontend code

fatimali03 committed:
feat: tenantid added

fatimali03 committed:
fix: package update

UmairKhan-Paysys commented on backend/.husky/pre-commit:
This should contain `npx lint-staged`

UmairKhan-Paysys commented on backend/jest.config.js:
[Ref](https://github.com/tazama-lf/event-director/blob/main/jest.config.ts)
The jest config has to have a threshold defined. The reference can be used.

Abdullah-Siddiqui-Paysys committed:
fix: integration changes as per 1.1 and 1.2 - v1

fatimali03 committed:
feat:  multi-field mapping implementation with all tests passing"

fatimali03 committed:
fix:knex file

fatimali03 committed:
fix: remove unnecessary code after table changes

fatimali03 committed:
feat: Define new API endpoints

fatimali03 committed:
feat: data model extension

fatimali03 committed:
fix:delete frontend folder

fatimali03 committed:
feat: dry run simulation of Schema and Mapping

fatimali03 committed:
fix: data-model extension fixed; and added endpoint path regeneration

fatimali03 committed:
feat: clone, edit and delete mapping

fatimali03 committed:
fix: config fixes

fatimali03 committed:
fix: clone and modify api

fatimali03 committed:
fix: simulation run

fatimali03 committed:
fix: common functionality moved to tcs-lib

fatimali03 committed:
feat: function to be used column added

fatimali03 committed:
feat: flowable implemented and sources removed from functions

MuhammadAli-Paysys committed:
feat: Cron Job Scheduler Module Added

MuhammadAli-Paysys committed:
feat: helpers added

MuhammadAli-Paysys committed:
feat: Get Schedule By Id Added

MuhammadAli-Paysys committed:
feat: Schedule Migration Added

fatimali03 committed:
fix: removed sources from function and corrected the three function names

fatimali03 committed:
feat: simulation/dry run service

fatimali03 committed:
fix: simulation error handling

fatimali03 committed:
fix: datamodel extension

fatimali03 committed:
fix: simulation fixed

fatimali03 committed:
fix: package file

fatimali03 committed:
feat: workflow implemented and removed data model extension

fatimali03 committed:
feat: approver approves and reject logic

fatimali03 committed:
fix: simulation fixed

fatimali03 committed:
fix: datamodel controller

fatimali03 committed:
feat: redis implemented for datamodel

MuhammadAli-Paysys committed:
feat: Databaser Service Integrated With PG

fatimali03 committed:
fix: simulation handles arrays now

fatimali03 committed:
feat: Add SUM and CONSTANT mapping transformations

- Added sumFields and constantValue to AddMappingDto interface
- Implemented SUM transformation for aggregating multiple numeric fields
- Implemented CONSTANT transformation for fixed value mappings

fatimali03 committed:
fix: approver claims added to simulation, added api for getting under-review endpoints

fatimali03 committed:
fix: simulation to handle arrays

fatimali03 committed:
fix; param name to be added with prefix redis

fatimali03 committed:
fix: redis added as prefix

Abdullah-Siddiqui-Paysys committed:
fix: transaction prefix added to tenantId param

MuhammadAli-Paysys committed:
feat: Job Module Added

fatimali03 committed:
feat: prefix added for mapping

fatimali03 committed:
fix: lint fixed

fatimali03 committed:
fix: prefix handling inside all types of mappings

fatimali03 committed:
fix: format fixed

fatimali03 committed:
fix: format fixed

fatimali03 committed:
fix: db fixed

MuhammadAli-Paysys committed:
feat: Status Api's Added

MuhammadAli-Paysys committed:
feat: Imports Updated

MuhammadAli-Paysys committed:
feat: Update Schedule Interface

MuhammadAli-Paysys committed:
feat: Activation Status Of Job

MuhammadAli-Paysys committed:
feat: Pull Job Creation

MuhammadAli-Paysys committed:
feat: Packages Installed

MuhammadAli-Paysys committed:
feat: env example updated

fatimali03 committed:
feat: session management added

fatimali03 committed:
fix; source for mapping will always be an array in db

fatimali03 committed:
fix: redis datamodel and remove transaction hsitory and added transaction details

fatimali03 committed:
fix: tazama data model

MuhammadAli-Paysys committed:
refactor: improve error handling and code structure in DryRunService and JobService

fatimali03 committed:
feat: mapping to show in transformed output

MuhammadAli-Paysys committed:
refactor: change id type from number to string in Schedule

fatimali03 committed:
fix: mapping made optional for simulation

fatimali03 committed:
Merge pull request #15 from tazama-lf/feat/paysys/fixes

fix: Bug Fixes

MuhammadAli-Paysys committed:
feat: add getConfig endpoint and implement SFTP configuration retrieval in DatabaseService

MuhammadAli-Paysys committed:
feat: update job status retrieval and implement SFTP config management in JobService

MuhammadAli-Paysys committed:
feat: add record_status field to queries in JobService

MuhammadAli-Paysys committed:
refactor: update transformFileToJSON return type and improve error handling

MuhammadAli-Paysys committed:
feat: add ConfigModule and ConfigService to JobModule and JobService tests

MuhammadAli-Paysys committed:
Merge branch 'feat/paysys/fixes' of https://github.com/tazama-lf/connection-studio into feat-paysys-enrichment

MuhammadAli-Paysys committed:
refactor: remove Flowable webhook controller and related tests

MuhammadAli-Paysys committed:
refactor: updated parsing services

MuhammadAli-Paysys committed:
feat: added modules to AppModule

fatimali03 committed:
fix: function added and config parsing fixed

fatimali03 committed:
fix: lint issues and schema error

fatimali03 committed:
fix: lint errors

fatimali03 committed:
fix: simulation fixed for fields added manually and display error for duplicate schema fields

fatimali03 committed:
fix: conflict resolved

fatimali03 committed:
fix: array handling for simulation

fatimali03 committed:
fix: transformed output

fatimali03 committed:
fix: simulation transformed output

fatimali03 committed:
fix; user issues and mapping type fixed

fatimali03 committed:
fix: transformations in mapping to be reflected in simulation output preview

fatimali03 committed:
fix: session activity

fatimali03 committed:
fix: xml simulation

fatimali03 committed:
fix: session management active user to remain logged in

fatimali03 committed:
fix: split transformation to show in simulation

fatimali03 committed:
fix: array payloads

fatimali03 committed:
Merge branch 'feat-paysys' into feat-paysys-enrichment

MuhammadAli-Paysys committed:
fix: update return type for createPull method and improve error handling

MuhammadAli-Paysys committed:
feat: add updateStatus Scheduler functionality

MuhammadAli-Paysys committed:
fix:  enhance update method with schedule status check

MuhammadAli-Paysys committed:
fix: streamline buffer initialization

fatimali03 committed:
fix: status maintainence

MuhammadAli-Paysys committed:
fix: update return types in query and dryRun methods for consistency

fatimali03 committed:
Merge branch 'feat-paysys' into feat-paysys-enrichment

MuhammadAli-Paysys committed:
refactor: migrate interfaces and types to @tazama-lf/tcs-lib and remove unused files

MuhammadAli-Paysys committed:
fix: update updateStatus method to require type and handle forbidden status

MuhammadAli-Paysys committed:
feat: add SFTP module with service and controller for file management

MuhammadAli-Paysys committed:
feat: implement Notify module and service for handling notifications

MuhammadAli-Paysys committed:
feat: add serialization interceptor and DTOs for pull job handling

MuhammadAli-Paysys committed:
feat: add readFile method to SftpService and update SftpController for file retrieval

MuhammadAli-Paysys committed:
feat: update job service to handle job status

MuhammadAli-Paysys committed:
feat: update .env.example to include SFTP configuration for consumer and producer

MuhammadAli-Paysys committed:
feat: job and scheduler services to include tenant ID handling and update SFTP file creation logic

MuhammadAli-Paysys committed:
fix: remove unnecessary tenant ID parameter from findOne method in getById

MuhammadAli-Paysys committed:
feat: enhance SFTP file handling with integrity checks and refactor file naming conventions

saeedhassansolangi commented on backend/src/config/config.service.ts:
```diff
@@ -0,0 +1,2346 @@
+    private readonly workflowService: ConfigWorkflowService,
+  ) {}
+
+  async createConfig(
```
`createConfig` function 
1. This function is too long, we can break this into smaller functions. 
2. we should avoid to log the entire payload

MuhammadAli-Paysys committed:
feat: add deleteFile method to SftpService and update related tests

MuhammadAli-Paysys committed:
fix: update schedule status check from APPROVED to DEPLOYED in JobService

fatimali03 committed:
fix: schema parsed payload

fatimali03 committed:
feat: exporter added

fatimali03 committed:
fix; Require anyClaim

fatimali03 committed:
fix;payload parsing

Abdullah-Siddiqui-Paysys committed:
Merge branch 'feat-paysys' of https://github.com/tazama-lf/connection-studio into feat-paysys

MuhammadAli-Paysys committed:
feat: refactor file name generation and add deleteFile call

fatimali03 committed:
fix: model

fatimali03 committed:
fix: .0 notation for mapping

fatimali03 committed:
Merge branch 'feat-paysys-enrichment' into feat-paysys

fatimali03 committed:
fix: lint issues and exporter and publisher addded
